python/tank/descriptor/io_descriptor/base.py

```diff
@@ -627,6 +630,12 @@ def _get_cache_paths(self):
         paths.append(self._get_primary_cache_path())
         return paths
 
+    def _get_temporary_cache_path(self):
+        """
+        Returns the temporary download cache path for this descriptor.
+        """
+        return os.path.join(self._bundle_cache_root, "tmp", self._download_identifier)
```

Comment:
Do we need return the same path consistently for an instance? If we do, please add to the docstring that this method will consistently return the same path for an instance. If not, i suggest consider making it a classmethod, returning a new temp cache path for every invocation. I haven't looked in detail, but i think if we can avoid storing the uuid as part of the instance state, that might end up being a simpler design.